dd-sdk-android-core/src/main/kotlin/com/datadog/android/api/feature/FeatureScopeExt.kt

```diff
@@ -0,0 +1,42 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.android.api.feature
+
+import com.datadog.android.core.metrics.TelemetryMetricType
+
+/**
+ * Measures the execution time for the given block and report it as a MethodCall telemetry metric.
+ * @param R the type of the result of the operation
+ * @param callerClass the class calling the measured method
+ * @param operationName the operationName to report in the metric
+ * @param samplingRate the sampling rate for the metric
+ * @param operation the operation to report
+ */
+fun <R : Any?> FeatureScope?.measureMethodCallPerf(
+    callerClass: Class<*>,
+    operationName: String,
+    samplingRate: Float = 100f,
+    operation: () -> R
+): R {
+    return if (this == null) {
+        operation()
+    } else {
+        val metric = startPerformanceMeasure(
+            callerClass = callerClass.name,
+            metric = TelemetryMetricType.MethodCalled,
+            samplingRate = samplingRate,
+            operationName = operationName
+        )
+
+        val result = operation()
+
+        val isSuccessful = (result != null) && ((result !is Collection<*>) || result.isNotEmpty())
```

Comment:
why generally speaking empty collection shouldn't be considered as a successful result? but probably for the methods we are going to cover with this call it makes sense.